src/Storage/Psr6CacheStorage.php

```diff
@@ -43,8 +55,16 @@ public function fetch($key)
      */
     public function save($key, CacheEntry $data)
     {
-        $item = $this->cachePool->getItem($key);
-        $item->set(serialize($data));
+        if ($this->lastItem && $this->lastItem->getKey() == $key) {
+            $item = $this->lastItem;
+        } else {
+            $item = $this->cachePool->getItem($key);
+        }
+
+        $this->lastItem = null;
```

Comment:
The reference to the last item is cleared after it is accessed, to avoid keeping a `CacheItem` in memory for longer than we need.